src/shared/tests.h

```diff
@@ -200,3 +200,42 @@ static inline int run_test_table(void) {
         DEFINE_TEST_MAIN_FULL(log_level, intro, NULL)
 #define DEFINE_TEST_MAIN(log_level)                     \
         DEFINE_TEST_MAIN_FULL(log_level, NULL, NULL)
+
+#define ASSERT_OK(expr)                                                                                  \
+        ({                                                                                               \
+                int _result = (expr);                                                                    \
+                if (_result < 0) {                                                                       \
+                        log_error_errno("Assertion failed: %s (result: %d, error: %m)", #expr, _result); \
+                        abort();                                                                         \
+                }                                                                                        \
+        })
+
+#define ASSERT_EQ(actual, expected)                                                                      \
+        ({                                                                                               \
+                int _actual = (actual);                                                                  \
+                int _expected = (expected);                                                              \
+                if (_actual != _expected) {                                                              \
+                        log_error("Assertion failed: %s != %s (actual: %d, expected: %d)", #actual, #expected, _actual, _expected); \
+                        abort();                                                                         \
+                }                                                                                        \
+        })
+
+#define ASSERT_GE(actual, expected)                                                                      \
```

Comment:
This is not taking in actual nor expected parameters, it's taking operands, so actual/expected are the wrong names for the variables and for the log message

Reply:
That's fine

Reply:
or op1 and op2